crawl-ref/source/initfile.cc

```diff
@@ -1480,6 +1481,83 @@ void game_options::add_feature_override(const string &text, bool prepend)
 #undef COL
     }
 }
+void game_options::remove_decayed_cloud_glyph(const string &text, bool prepend)
+{
+    string key = lowercase_string(text);
+    cloud_decay_state cd_key;
+
+    if (key=="terminal")
+        cd_key = CD_TERMINAL;
+    else if (key=="weak")
+        cd_key = CD_WEAK;
+    else if (key=="fading")
+        cd_key = CD_FADING;
+    else
+    {
+        report_error("Bad key in option decayed_cloud_glyphs: \"%s\"",
+                     key.c_str());
+        return;
+    }
+
+    auto i = decayed_cloud_glyphs.find(cd_key);
+    if (i == decayed_cloud_glyphs.end())
+    {
+        report_error("Attempt to remove absent value in option "
+                     "decayed_cloud_glyphs: \"%s\"", text.c_str());
+    }
+    else
+        decayed_cloud_glyphs.erase(i);
+}
+
+void game_options::add_decayed_cloud_glyph(const string &text, bool prepend)
+{
+    vector<string> mapping = split_string(":", text);
+    if (mapping.size() != 2)
+    {
+        report_error("Couldn't parse field in option "
+                     "decayed_cloud_glyphs: \"%s\"", text.c_str());
+    }
+    else
+    {
+        cglyph_t g;
+        string key = lowercase_string(mapping[0]);
+        cloud_decay_state cd_key;
+
+        if (key=="terminal")
+            cd_key = CD_TERMINAL;
+        else if (key=="weak")
+            cd_key = CD_WEAK;
+        else if (key=="fading")
+            cd_key = CD_FADING;
+        else
```

Comment:
It's probably slightly neater to use `switch ... case "terminal: ... default:` here, but it's a matter of preference.

Reply:
Except that you cannot switch over strings :)

Reply:
Oh, right :)

Reply:
Although I did look over the commit a bit more and ther is a new enum introduced. Dedicated to string / from string functions instead of ad hoc conersion would greatly improve the PR. Does that make sense or should I elaborate on that?